### PR TITLE
[vcpkg script] Normalize PATH so that insertion of msys paths always works

### PR DESCRIPTION
**Pull request by Neumann-A**
This is required if the PATH variable has been modified and doesn't follow the normal windows style

### CODE REVIEW

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 1]
```diff
@@ -229,15 +229,28 @@ function(vcpkg_configure_make)
...
-            string(REPLACE ";$ENV{SystemRoot}\\System32;" "${append_env};${MSYS_ROOT}/usr/bin;$ENV{SystemRoot}\\System32;" NEWPATH "$ENV{PATH}")
-            string(REPLACE ";$ENV{SystemRoot}\\system32;" "${append_env};${MSYS_ROOT}/usr/bin;$ENV{SystemRoot}\\system32;" NEWPATH "$ENV{PATH}")
-            set(ENV{PATH} "${NEWPATH}")
+            cmake_path(CONVERT "$ENV{PATH}" TO_CMAKE_PATH_LIST PATH_LIST NORMALIZE)
+            cmake_path(CONVERT "$ENV{SystemRoot}" TO_CMAKE_PATH_LIST SYSTEM_ROOT)
+
+            list(FIND PATH_LIST "${SYSTEM_ROOT}/system32/" index)
+            if(index EQUAL "-1")
+                list(FIND PATH_LIST "${SYSTEM_ROOT}/System32/" index)
+            endif()
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_configure_make.cmake:243]: Did anyone else ever notice the case mismatch?
~~~
ENV{SystemRoot}: C:\WINDOWS
ENV{PATH}:   ...;C:\Windows\...

Comment (dg0yt) [reply scripts/cmake/vcpkg_configure_make.cmake:243]: > Did anyone else ever notice the case mismatch?
> 
> ```
> ENV{SystemRoot}: C:\WINDOWS
> ENV{PATH}:   ...;C:\Windows\...
> ```

Hm, actually only in MSYS2's `bash`.

Comment (dg0yt) [inline scripts/cmake/vcpkg_configure_make.cmake:243]: AFAICT the trailing slash is wrong.

Comment (Neumann-A) [reply scripts/cmake/vcpkg_configure_make.cmake:243]: Means `cmake_path(CONVERT "$ENV{SystemRoot}" TO_CMAKE_PATH_LIST SYSTEM_ROOT)` also needs `NORMALIZE`

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 2]
```diff
@@ -229,15 +229,43 @@ function(vcpkg_configure_make)
             debug_message("Using make triplet: ${arg_BUILD_TRIPLET}")
         endif()
         if(CMAKE_HOST_WIN32)
-            set(append_env)
+            vcpkg_list(SET add_to_env)
             if(arg_USE_WRAPPERS)
-                set(append_env ";${MSYS_ROOT}/usr/share/automake-1.16")
-                string(APPEND append_env ";${SCRIPTS}/buildsystems/make_wrapper") # Other required wrappers are also located there
+                vcpkg_list(APPEND add_to_env "${MSYS_ROOT}/usr/share/automake-1.16")
+                vcpkg_list(APPEND add_to_env "${SCRIPTS}/buildsystems/make_wrapper") # Other required wrappers are also located there
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_make.cmake:235]:
```suggestion
                vcpkg_list(APPEND add_to_env "${SCRIPTS}/buildsystems/make_wrapper") # Other required wrappers are also located there
                vcpkg_list(APPEND add_to_env "${MSYS_ROOT}/usr/share/automake-1.16")
```

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
> This is required if the PATH variable has been modified and doesn't follow the normal windows style

Can you clarify what a broken example looks like? I think this change is fine but I admit to not having the best understanding of the problem.

**Review by BillyONeal — APPROVED** [review]

**Comment by Neumann-A** [comment]
> Can you clarify what a broken example looks like? I think this change is fine but I admit to not having the best understanding of the problem.

`C:/Windows/System32` instead of `C:\Windows\System32` which you get quite fast if you use cmake to set `PATH` in the triplet

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**